tests/trnsysGUI/connection/testCleanConnector.py

```diff
@@ -0,0 +1,221 @@
+            patchString="NiceConnectorFromBelow",
+            addItemCount=1,
+            fromPortSide=1,
+            toPortSide=None,
+            nrOfNewItems=5,
+            nSegsExpected=3,
+            xStartCoordsExpected=[0.0, 0.0, 100.0],
+            xEndCoordsExpected=[0.0, 100.0, 100.0],
+            yStartCoordsExpected=[0.0, -15.666, -15.666],
+            yEndCoordsExpected=[-15.666, -15.666, 100.0],
+        )
+
+    def testDPconnectionUsingFromBelowConnectorWithOtherPortSide(self):
+        self._runNiceConnWithTests(
+            connectorType=_gnc.NiceConnectorFromBelow,
+            patchString="NiceConnectorFromBelow",
+            operation="add",
+            addItemCount=1,
+            fromPortSide=3,
+            toPortSide=None,
+            nrOfNewItems=5,
+            nSegsExpected=3,
+            xStartCoordsExpected=[0.0, 0.0, 100.0],
+            xEndCoordsExpected=[0.0, 100.0, 100.0],
+            yStartCoordsExpected=[0.0, 15.666, 15.666],
+            yEndCoordsExpected=[15.666, 15.666, 100.0],
+        )
+
+    def _runNiceConnWithTests(  # pylint: disable = too-many-arguments, too-many-locals
+        self,
+        connectorType,
+        addItemCount,
+        fromPortSide,
+        nSegsExpected,
+        nrOfNewItems,
+        patchString,
+        toPortSide,
+        xEndCoordsExpected,
+        xStartCoordsExpected,
+        yEndCoordsExpected,
+        yStartCoordsExpected,
+        operation="subtract",
+    ):
+
+        dpConnection, rad, segmentItemFactory = self._getSetup()
+        mockMethod = self._runNiceConn(
+            connectorType, dpConnection, patchString, rad, segmentItemFactory, fromPortSide, operation
+        )
+
+        assert dpConnection.fromPort.createdAtSide == fromPortSide
+        if toPortSide:
+            assert dpConnection.toPort.createdAtSide == toPortSide
+        assert mockMethod.call_count == addItemCount
+        newGraphicalItems = mockMethod.call_args[0][0]
+        assert len(newGraphicalItems) == nrOfNewItems
+
+        self._checkItemType(nSegsExpected, newGraphicalItems)
+
+        xStartCoords, yStartCoords, xEndCoords, yEndCoords = self._getLineCoords(newGraphicalItems[0:nSegsExpected])
+
+        assert xStartCoords == xStartCoordsExpected
+        assert yStartCoords == yStartCoordsExpected
+        assert xEndCoords == xEndCoordsExpected
+        assert yEndCoords == yEndCoordsExpected
+
+    @staticmethod
+    def _getSetup():
+        rad = 4
+        dpConnection = _fc.createFullDoublePipeConnectionMock("dpConnectionName")
+        segmentItemFactory = FakeSegmentItemFactory(dpConnection)
+        return dpConnection, rad, segmentItemFactory
+
+    @staticmethod
+    def _runNiceConn(
+        connectorType, dpConnection, patchString, rad, segmentItemFactory, fromPortSide=None, operation="subtract"
+    ):
+        fullPatchString = "trnsysGUI.connection.getNiceConnector." + patchString + "._addGraphicsItems"
```

Comment:
can we patch the object directly, without using strings?
Does the spy still exist then?